Manager/PostManager.php

```diff
@@ -114,4 +114,19 @@ public function getThumbnailPath(Post $post)
 
         return $post->getGuid();
     }
+    
+    /**
+     * @param \DateTime $date
+     * @return array
+     */
+    public function findByDate(\DateTime $date)
+    {
+        $qb = $this->repository->createQueryBuilder('c');
```

Comment:
Can you please just call here a repository method and move this query logic into the repository class?

This way, `PostManager` should have the following code:

```php
    /**
     * Returns posts for current date or a specified date.
     *
     * @param \DateTime $date
     *
     * @return array
     */
    public function findByDate(\DateTime $date = null)
    {
        $date = $date ?: new \DateTime();

        return $this->repository->findByDate($date);
    }
```

and move this query logic into `PostRepository`:

```php    
    /**
     * Returns posts for current date or a specified date.
     *
     * @param \DateTime $date
     *
     * @return array
     */
    public function findByDate(\DateTime $date)
    {
        $qb = $this->createQueryBuilder('c')
            ->select('c')
            ->where('c.date = :date')
            ->addOrderBy('c.id')
            ->setParameter('date', '%' . $date->format('Y-m-d') . '%');

        return $qb->getQuery()->getResult();
    }
```